burn-core/src/nn/initializer.rs

```diff
@@ -10,74 +10,116 @@ use crate as burn;
 /// Enum specifying with what values a tensor should be initialized
 #[derive(Config, Debug, PartialEq)]
 pub enum Initializer {
-    /// Fills tensor with values drawn uniformly between specified values
-    Uniform(f64, f64),
-    /// Must be implemented by caller. TODO change to NormalizedUniform, fills tensor with values drawn uniformly between -sqrt(1/fan_in) and sqrt(1/fan_in).
-    UniformDefault,
-    /// Fills tensor with values drawn from normal distribution with specified mean and std
-    Normal(f64, f64),
     /// Fills tensor with specified value everywhere
     Constant(f64),
     /// Fills tensor with 1s everywhere
     Ones,
     /// Fills tensor with 0s everywhere
     Zeros,
+    /// Fills tensor with values drawn uniformly between specified values
+    Uniform(f64, f64),
```

Comment:
I think it would be valuable to add names to each argument when applicable.

```rust
pub enum Initializer {
    Uniform {
        min: f64,
        max: f64,
     },
     Normal {
         mean: f64,
         std: f64,
     } ,
     ...
}
```

I think it would help knowing what represent the value of each argument.

Reply:
I personally like tuples for the initialization. It's cleaner for the configuration like this: 
```rust
#[derive(Config)]
pub struct LstmConfig {
    /// Lstm initializer
    #[config(default = "Initializer::XavierNormal(1.0)")]
    pub initializer: Initializer,
}
```
With the named struct, you need extra braces and param name + `=`.

I vote we leave it.